applications/tari_merge_mining_proxy/src/main.rs

```diff
@@ -388,6 +424,50 @@ fn get_spending_key() -> (PrivateKey, PrivateKey) {
     (key, r)
 }
 
+// Select lowest difficulty
+fn select_difficulty(data: &[u8], tari_difficulty: u64) -> Vec<u8> {
```

Comment:
nit: Just a style thing, but since this method alters `data` and then returns it, I would say it's better to `move` `data` in as a `Vec<u8>` so that the old version is not used by code elsewhere. The name of the method also implies that it is only reading and not altering the data. A better name might be: `select_and_update_lowest_difficulty`